lib/AST/Decl.cpp

```diff
@@ -6308,10 +6308,17 @@ bool ClassDecl::isForeignReferenceType() const {
   if (!clangRecordDecl)
     return false;
 
-  CxxRecordSemanticsKind kind = evaluateOrDefault(
-      getASTContext().evaluator,
-      CxxRecordSemantics({clangRecordDecl, getASTContext()}), {});
-  return kind == CxxRecordSemanticsKind::Reference;
+  // DOUBT: How can I call evaluateOrDefault with CxxRecordSemanticsKind without
```

Comment:
We need `ClangImporter::Implementation` for the diagnostics right? We cannot access it here, but also we might not want to trigger diagnostics when this function is called. So I think one way out is to make `ClangImporter::Implementation` optional (a pointer) and only emit diagnostics when it is actually provided. So we can pass a nullpointer here in desc, but a valid pointer in every other case. In this case, we need both a reference to ASTContext and a pointer to the implementation object.